src/click/testing.py

```diff
@@ -380,18 +380,25 @@ def invoke(
         )
 
     @contextlib.contextmanager
-    def isolated_filesystem(self):
+    def isolated_filesystem(self, tempdir=None):
         """A context manager that creates a temporary folder and changes
         the current working directory to it for isolated filesystem tests.
+
+        :param tempdir: the path where the temporary directory should be created
+
+        .. versionchanged:: 7.1.2
         """
+        tempfile.tempdir = tempdir
```

Comment:
This modifies global state and doesn't reset it. It's also not needed, `mkdtemp` takes a `dir` parameter.